scripts/benchmark.sh

```
@@ -91,9 +105,20 @@ stop_server() {
   done
 }
 
+# Every abnormal exit — a failed readiness check, a `set -e` trip, Ctrl-C —
+# would otherwise leave the current section's server group running.
+trap stop_server EXIT INT TERM
+
 start_server() {
   DJANGO_BOLT_WORKERS=$WORKERS $SETSID_BIN uv run python manage.py runbolt --host $HOST --port $PORT --processes $P >/dev/null 2>&1 &
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)